library/bigip_software.py

```diff
@@ -212,673 +290,1060 @@
+        err = 'Could not execute command. Most likely device is unresponsive.'
+        result = self.client.api.tm.util.bash.exec_cmd('run', utilCmdArgs=cmd)
+        if result:
+            return result
+        else:
+            raise F5ModuleError(err)
 
-        images = self.read()
-        total = sum(len(v) for v in images.itervalues())
+    def image_exists_on_device(self):
+        if self.check_product_info():
+            return self.image_exists_on_device_by_name()
+        images = self.list_images_on_device()
+        for image in images:
+            if image.version == self.want.version:
+                self.name_check(image.name)
+                return True
+        return False
 
-        if hotfix and software:
-            if not self.is_software_available(software) and not self.is_hotfix_available(hotfix):
-                return False
-        elif hotfix:
-            if not self.is_hotfix_available(hotfix):
-                return False
-        elif software:
-            if not self.is_software_available(software):
-                return False
+    def hotfix_exists_on_device(self):
+        if self.check_product_info():
+            return self.hotfix_exists_on_device_by_name()
+        hotfixes = self.list_hotfixes_on_device()
+        for hotfix in hotfixes:
+            if hotfix.version == self.want.version:
+                if hotfix.build == self.want.build:
+                    self.name_check(hotfix.name, True)
+                    return True
+        return False
 
-        if hotfix:
-            self.delete(hotfix)
-            total -= 1
+    def hotfix_exists_on_device_by_name(self):
+        return self.client.api.tm.sys.software.hotfix_s.hotfix.exists(
+            name=self.want.hotfix_name
+        )
 
-        if software:
-            self.delete(software)
-            total -= 1
+    def image_exists_on_device_by_name(self):
+        return self.client.api.tm.sys.software.images.image.exists(
+            name=self.want.software_name
+        )
+    def load_volume_on_device(self):
+        volume = self.client.api.tm.sys.software.volumes.volume.load(
+            name=self.want.volume)
+        return volume
+
+    def reboot_volume_on_device(self):
+        self.client.api.tm.sys.software.volumes.exec_cmd(
+            'reboot', volume=self.want.volume
+            )
+    def volume_exists_on_device(self):
+        return self.client.api.tm.sys.software.volumes.volume.exists(
+            name=self.want.volume
+        )
 
-        self.wait_for_images(total)
 
-        return True
+class ArgumentSpec(object):
+    def __init__(self):
+        self.states = ['absent', 'activated', 'installed', 'present']
+        self.supports_check_mode = True
+        self.argument_spec = dict(
+            state=dict(
+                default='activated',
+                choices=self.states
+            ),
+            force=dict(
+                required=False,
```

Comment:
the comment was for line 1283. github shows a small blob above the comment which i assume you're referring to here